### PR TITLE
Regex for regkey in God mode too lenient and potentially dangerous #4366

### LINKED ISSUE #4366: Regex for regkey in God mode too lenient and potentially dangerous
Similar to #4361.
In particular, the third one: `"([a-zA-Z0-9-_]){50,}"`, the ones where `Const.ParamsNames.REGKEY` is involved are okay. It means that _all_ combinations of alphabets, numbers, -, and _ of length 50 and more, wrapped around double quotes, will be replaced.
It does replace the regkey wrapped in double quotes, but there are so many more stuffs that are replaced by this. This time there are more unwanted changes.

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #4366 and closes #4367 as the demo is no longer needed.
The additionalInfoId and the super long name that are erroneously replaced before are now recovered.
There are some indentation fixes which I think is a beneficial side effect, but if it's better to leave them I'll just undo the last commit (edit: now 2nd last commit).

### CODE REVIEW

File: src/test/resources/pages/instructorHomeCourseDeleteSuccessful.html [hunk 1]
```diff
@@ -1,132 +1,125 @@
...
-                                 Search
-                              </button>
-                           </span>
-                        </div>
-                        <input name="searchstudents" type="hidden" value="true"/>
-                                                <input name="searchcommentforstudents" type="hidden" value="false"/>
-                                                <input name="searchcommentforresponses" type="hidden" value="false"/>
-                                                <input name="user" type="hidden" value="CHomeUiT.instructor.tmms"/>
-                                             </form>
+   <div class="container" id="mainContent">
```
Comment (thyageshm) [inline src/test/resources/pages/instructorHomeCourseDeleteSuccessful.html]: Hey, any reason why this html has changed from a full html check to mainContent check?

**Comment by thyageshm** [comment]
good catch on the problem of questionid check! Everything is good :-) just a minor clarification. Once that is clarified, this one should be good to go!

File: src/test/resources/pages/instructorHomeCourseDeleteSuccessful.html (hunk 1 above)
Comment (wkurniawan07) [inline src/test/resources/pages/instructorHomeCourseDeleteSuccessful.html]: This is not that type of change. This happens as a remnant of #4324.

Comment (thyageshm) [inline src/test/resources/pages/instructorHomeCourseDeleteSuccessful.html]: I see,ok!

**Pull request merged**